### ISSUE TITLE
Degree in a graded module

### ISSUE THREAD

**Issue by JohnAAbbott** [oscar-system/Oscar.jl#2344]
**Describe the bug**
I had expected the `degree` function to work on elements of a graded module.  Instead it throws an "inscrutable" error.

**To Reproduce**
```
using Oscar
# Copied from issue #2335, example by W Decker:
S, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"], [1 0 1; 0 1 1]);

GG = free_abelian_group(2);
FM_grad = graded_free_module(S, GG.([[1, 2], [-1, 3]]));
(ee1,ee2) = gens(FM_grad);

degree(ee1)  # --> ERROR: LoadError: Cannot coerce elements into common structure
```

**Expected behavior**
I expected to get `[1, 2]` (or equivalent) since that is the degree declared when creating the graded module.

**System (please complete the following information):**
Please paste the output of `Oscar.versioninfo(full=true)` below. If this does
not work, please paste the output of Julia's `versioninfo()` and your Oscar
version.
```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.12.1-DEV - #JAA/exterior_algebra, 483ba031 -- 2023-05-04 16:24:43 +0200
  combining:
    AbstractAlgebra.jl   v0.29.4
    GAP.jl               v0.9.4
    Hecke.jl             v0.18.9
    Nemo.jl              v0.33.7
    Polymake.jl          v0.9.0
    Singular.jl          v0.18.2
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.5+0
    GAP_jll                 v400.1200.200+0
    Singular_jll            v403.201.1+0
    libpolymake_julia_jll   v0.9.0+0
    libsingular_julia_jll   v0.30.0+0
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 20 × Intel(R) Core(TM) i9-10900 CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 20 virtual cores
Environment:
  JULIA_LOAD_PATH = @:@v#.#:@stdlib:/environments/globalenv
  JULIA_DEPOT_PATH = /home/zez25zuh/.julia:
Official https://julialang.org/ release
```

**Additional context**
None.

**Comment by jankoboehm** [comment]
The example does not make sense, since creating a graded polynomial ring creates a grading group, which has to be also used for the module. So the expectation is not returning the degree but an error. However, there seems to be missing somewhere a check which should error already in the constructor, could be that it was added in the PR waiting. Will check, thanks for catching that!

**Comment by JohnAAbbott** [comment]
Thanks for the clarification!

**Comment by wdecker** [comment]
addressed in[77522f6](https://github.com/oscar-system/Oscar.jl/pull/2334/commits/77522f6e6903b2e5ae42b037be6efcc495da9053)
